### PR TITLE
#BN-23:Buyer should be able to clear their cart

### PR DESCRIPTION
**Pull request by uwdjany**
# PR Description
https://trello.com/c/KWfchsBs/23-buyer-should-be-able-to-clear-their-cart
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] The buyer's cart should be emptied
- [x] The cart total should be reset to zero
- [x] A cart cleared confirmation message should be sent to the frontend to be displayed to the buyer
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `npm i` to install all required dependencies
2. run `npm run dev`
3. step 3
4. step 4...
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: src/controllers/product/clearCartController.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+const clearCart = async (req, res) => {
+  try {
+    // const buyer = req.user?.id; // using optional chaining operator to avoid undefined error
+    // if (!buyer) {
+    //   return res.json({
+    //     status: 400,
+    //     message: 'User not found',
+    //   });
+    // }
```
Comment (kunda4) [inline src/controllers/product/clearCartController.js]: Remove this commented Line

File: src/database/models/cart.js [hunk 2]
```diff
@@ -15,8 +15,8 @@ module.exports = (sequelize, DataTypes) => {
     {
       user_id: DataTypes.STRING,
       product_id: DataTypes.STRING,
-      quantity: DataTypes.INTEGER,
-      date_added: DataTypes.STRING,
+      // quantity: DataTypes.INTEGER,
+      //date_added: DataTypes.STRING,
```
Comment (kunda4) [inline src/database/models/cart.js]: Remove this too

**Review by kunda4 — COMMENTED** [review]
Well done djanati just for making code look great remove that commented line

File: src/database/models/cart.js (hunk 2 above)
Comment (uwdjany) [reply src/database/models/cart.js]: Done!

**Review by kunda4 — DISMISSED** [review]

**Review by dushimeemma — DISMISSED** [review]
Your commit message should match standard.
example on [wiki](https://github.com/atlp-rwanda/engineering-playbook/wiki)

**Review by dushimeemma — CHANGES_REQUESTED** [review]
Your commit message should match standard.
example on [wiki](https://github.com/atlp-rwanda/engineering-playbook/wiki)

**Review by Niyobuzima — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]

**Pull request merged**